privacyidea/lib/eventhandler/webhookevanthandler.py

```diff
@@ -0,0 +1,150 @@
+
+# License:  AGPLv3
+# (c)
+#
+# This code is free software; you can redistribute it and/or
+# modify it under the terms of the GNU AFFERO GENERAL PUBLIC LICENSE
+# License as published by the Free Software Foundation; either
+# version 3 of the License, or any later version.
+#
+# This code is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU AFFERO GENERAL PUBLIC LICENSE for more details.
+#
+# You should have received a copy of the GNU Affero General Public
+# License along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+#
+__doc__ = """This is th event handler module for posting webhooks.
+You can send an webhook to trigger an event on an other system or use to replace
+api requests and reduce your traffic this way.
+
+"""
+from privacyidea.lib.eventhandler.base import BaseEventHandler
+from privacyidea.lib import _
+import json
+import logging
+import requests
+from requests.exceptions import HTTPError, Timeout, ConnectionError, RequestException
+from collections import defaultdict
+
+log = logging.getLogger(__name__)
+
+
+class CONTENT_TYPE(object):
+    """
+    Allowed type off content
+    """
+    JSON = "json"
+    URLENCODED = "urlendcode"
+
+
+class ACTION_TYPE(object):
+    """
+    Allowed actions
+    """
+    POST_WEBHOOK = "post_webhook"
+
+
+class WebHookHandler(BaseEventHandler):
+    """
+    With the WebHook Handler, a webhook can be sent at a particular event
+    """
+
+    identifier = "WebHookHandler"
+    description = "This eventhandler can post webhooks"
+
+    @property
+    def allowed_positions(cls):
+        """
+        This returns the allowed positions of the event handler definition.
+
+        :return: list of allowed positions
+        """
+        return ["post", "pre"]
+
+    @property
+    def actions(cls):
+        """
+        This method returns a dictionary of allowed actions and possible
+        options in this handler module.
+
+        :return: dict with actions
+        """
+        # The event handler has just one action. Maybe we can  hide action select for the clarity of the UI
+        actions = {ACTION_TYPE.POST_WEBHOOK: {
+            "URL": {
+                "type": "str",
+                "required": True,
+                "description": _("The URL the WebHook is posted to")
+            },
+            "content_type": {
+                "type": "str",
+                "required": True,
+                "description": _("The encoding that is sent to the WebHook, for example json"),
+                "value": [
+                    CONTENT_TYPE.JSON,
+                    CONTENT_TYPE.URLENCODED]
+            },
+            "data": {
+                "type": "str",
+                "required": True,
+                "description": _("The data posted in the WebHook")
+            }
+        }}
+        return actions
```

Comment:
Please add a simple test, that calls the property `actions` to verify the availability.

Reply:
https://github.com/privacyidea/privacyidea/pull/3154/commits/63296154129247c5d0cf0c73c7119668bd1a02f0